contracts/ERC721Collateralizer.sol

```diff
@@ -0,0 +1,372 @@
+/*
+
+  Copyright 2017 Dharma Labs Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity 0.4.18;
+
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+import "zeppelin-solidity/contracts/token/ERC721/ERC721.sol";
+
+import "./TermsContract.sol";
+import "./DebtRegistry.sol";
+import "./ERC721TokenRegistry.sol";
+import {PermissionsLib, PermissionEvents} from "./libraries/PermissionsLib.sol";
+
+
+/**
+  * Contains functionality for collateralizing NFTs, by transferring them from
+  * a debtor address to this contract as a custodian.
```

Comment:
I would add a `NOTE` in here that the `ERC721Collateralizer` only operates with a trusted set of `termsContracts` -- "untrusted" terms contracts are not able to collateral assets in it